test.sh

```diff
@@ -6,8 +6,37 @@
 Me=$(basename "$0")
 set -euo pipefail
 
-# location of binaries
-BINDIR="${BINDIR:-bin}"
+# location of binaries, which live under $BUILD_DIR, if set.
+build_dir="${BUILD_DIR:-build}"
+BINDIR="${BINDIR:-${build_dir}/bin}"
+
+# Location of binaries, which live under $BUILD_DIR, if BINDIR is not set.
+# If BINDIR -was- set in the user's env, we need to drive off of that.
+# To avoid raising a script error by referencing this variable if it was
+# -not-set- we just initialized it above to this value. Thus, if the
+# following is true, it means the env-var BINDIR was -not- set already.
+if [ "${BINDIR}" == "${build_dir}/bin" ]; then
```

Comment:
This check "Is BINDIR set in the env-var? Otherwise, synthesize it here ..." and the preceding comment block is a bit tacky. Suggestions welcome to rework that, but I think it's good enough the way it is now.